cloudformation/template.go

```diff
@@ -16,32 +16,44 @@ type Template struct {
 	Transform                *Transform             `json:"Transform,omitempty"`
 	Description              string                 `json:"Description,omitempty"`
 	Metadata                 map[string]interface{} `json:"Metadata,omitempty"`
-	Parameters               map[string]interface{} `json:"Parameters,omitempty"`
+	Parameters               Parameters             `json:"Parameters,omitempty"`
 	Mappings                 map[string]interface{} `json:"Mappings,omitempty"`
 	Conditions               map[string]interface{} `json:"Conditions,omitempty"`
 	Resources                Resources              `json:"Resources,omitempty"`
-	Outputs                  map[string]interface{} `json:"Outputs,omitempty"`
+	Outputs                  Outputs                `json:"Outputs,omitempty"`
 }
 
 type Parameter struct {
-	Type                  string   `json:"Type"`
-	Description           string   `json:"Description,omitempty"`
-	Default               string   `json:"Default,omitempty"`
-	AllowedPattern        string   `json:"AllowedPattern,omitempty"`
-	AllowedValues         []string `json:"AllowedValues,omitempty"`
-	ConstraintDescription string   `json:"ConstraintDescription,omitempty"`
-	MaxLength             int      `json:"MaxLength,omitempty"`
-	MinLength             int      `json:"MinLength,omitempty"`
-	MaxValue              float64  `json:"MaxValue,omitempty"`
-	MinValue              float64  `json:"MinValue,omitempty"`
-	NoEcho                bool     `json:"NoEcho,omitempty"`
+	Type                  string      `json:"Type"`
+	Description           string      `json:"Description,omitempty"`
+	Default               interface{} `json:"Default,omitempty"`
```

Comment:
It have to be changed to Interface from string because there can be for example:

Parameters:
  TimeoutParam:
    Type: Number
    Default: 10